Assets/Scripts/Gameplay/Powerup/PowerupObjectBase.cs

```diff
@@ -5,27 +5,40 @@ namespace CT6RIGPR
 
     public abstract class PowerupObjectBase : MonoBehaviour
     {
-        bool _hasBeenPickedUp = false;
+        private bool _hasBeenPickedUp = false;
         private AudioSource _audioSource;
         [SerializeField] private CollectableData _collectableData;
         [SerializeField] private GameObject _powerupVisualPrefab;
+        [SerializeField] private MeshRenderer _powerupMeshRenderer;
 
         /// <summary>
         /// The visual prefab of the powerup. For use with inventory.
         /// </summary>
         public GameObject PowerupVisualPrefab => _powerupVisualPrefab;
 
+        /// <summary>
+        /// Bool for whether the powerup has been picked up.
+        /// </summary>
+        public bool HasBeenPickedUp => _hasBeenPickedUp;
+
         private void Start()
         {
             _audioSource = GetComponent<AudioSource>();
         }
 
+        public void ResetPowerUp()

```

Comment:
Missing documentation

Reply:
Have added a description to that method.